### DISCUSSION TITLE
osu!taiko Touch Device Mod Handling

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#19307] [Q&A] [answered]
Now that we have touch support (thanks, by the way :heart:), scores using the touch UI should submit with the Touch Device mod.

What I'm not sure of, however, is how mouse plays should be qualified. In my opinion, legacy mouse input doesn't count as TD, but using this new touch support with the mouse does count as TD. ~~I know the community is already wary of "convert mono cheesing" with this new input system, so gathering community opinion is of great importance.~~ (See Lawtrohux's answer)

I'll be gathering taiko tournament community feedback on this implementation and opening candidate discussions where appropriate :+1:

**Comment by Lawtrohux** [comment]
The osu!taiko touch device implementation has each input key frozen until released - it behaves exactly like a physical keyboard without any rebound time. So in regards to your point on difficulty and performance, it shouldn't be much of an issue.

**Comment by frenzibyte** [comment, marked as answer]
https://github.com/ppy/osu/issues/5378

**Reply by zachmanthethird** [reply]
~~Does this also apply to osu!web? TD isn't showing on my recent score which I played on my phone.~~
Oh it's coupled with #12548. Then yes, though that issue could use some clarification on scope, as it advertises itself as a lazer-only issue.

**Reply by frenzibyte** [reply]
Yes this applies to osu!web, because the client doesn't signal to the server that the user is playing with a touch device.
